src/main/java/com/palantir/gradle/versions/VersionsPropsPlugin.java

```diff
@@ -92,6 +92,15 @@ public final void apply(Project project) {
                     });
             project.getTasks().named("check").configure(task -> task.dependsOn(checkNoUnusedConstraints));
 
+            TaskProvider<CheckOverbroadConstraints> checkBadPins = project.getTasks()
```

Comment:
Change this so we only make the task/have the `check.dependsOn` if there is a gradle property set to true (use `project.findProperty("gcv.overbroad.constraints")` - will need to add to the tests by making a `gradle.properties` file in the setup method and adding `gcv.overbroad.constraints=true`)